ocp_resources/resource.py

```diff
@@ -134,9 +133,27 @@ def sub_resource_level(current_class: Any, owner_class: Any, parent_class: Any)
     return None
 
 
-# Exceptions classes
+def change_dict_value_to_hashed(resource_dict: Dict[Any, Any], key_name: str) -> Dict[Any, Any]:
+    """
+    Recursively search a nested dictionary for a given key and changes its value to "******" if found.
+
+    Args:
+        resource_dict: The nested dictionary to search.
+        key_name: The key to find.
 
-# End Exceptions classes
+    Returns:
+        The modified dictionary.
+    """
+    if isinstance(resource_dict, dict):
+        for key, value in resource_dict.items():
+            if key == key_name:
+                resource_dict[key] = "******"
+            elif isinstance(value, dict):
+                resource_dict[key] = change_dict_value_to_hashed(value, key_name)
+            elif isinstance(value, list):
```

Comment:
```suggestion
                    elif isinstance(value, list) and any(isinstance(item, dict) for item in value):
```

to enter only if we have dict under the list

Reply:
I don't think it would be needed. 
```
            elif isinstance(value, list):
                for key_list, value_list in enumerate(value):
                    value[key_list] = change_dict_value_to_hashed(value_list, key_name)
```
This call `change_dict_value_to_hashed(value_list, key_name)` on value will simply return the value_list, as value_list is a list. and change_dict_value_to_hashed() would only work on dicts.

But may be I should add a check for list in line 156? What do you think?

Reply:
`But may be I should add a check for list in line 156? What do you think?` 
I don;t understand.

Reply:
To ensure recursion handles lists that are actually list I need to update this code. I was thinking
```
       if isinstance(resource_dict, dict):
        for key, value in resource_dict.items():
            if key == key_name:
                resource_dict[key] = "******"
            elif isinstance(value, dict):
                resource_dict[key] = change_dict_value_to_hashed(value, key_name)
            elif isinstance(value, list):
                for key_list, value_list in enumerate(value):
                    value[key_list] = change_dict_value_to_hashed(value_list, key_name)
    if isinstance(resource_dict, list):
        for key_list, value_list in enumerate(resource_dict):
            if isinstance(value_list, dict):
                resource_dict[key_list] = change_dict_value_to_hashed(value_list, key_name)
    return resource_dict
```